src/ion_data/mod.rs

```diff
@@ -1,24 +1,21 @@
 mod ion_eq;
 mod ion_ord;
 
-use crate::{Element, IonResult};
 use std::cmp::Ordering;
 use std::fmt::{Display, Formatter};
 
-use crate::element::{Annotations, Value};
-use delegate::delegate;
 pub use ion_eq::IonEq;
 pub(crate) use ion_ord::IonOrd;
```

Comment:
Something I think we should consider (and I am okay just opening an issue for this) is whether we actually need a public `IonEq` trait if `IonData` solves this.  The use case of `IonEq` is that users can apply Ion equality without conflation, `IonData` formalizes that relationship with order as well (and maybe eventually hashing).

I also think this has another benefit--end users cannot implement `IonEq` outside the crate--I see this as an effective reduction in API surface.